### DISCUSSION TITLE
prometheus-k8s route is returning 404 after OKD cluster upgrade to 4.13

### DISCUSSION THREAD

**Discussion by imdmahajankanika** [okd-project/okd#1698] [General]
After the cluster upgrade, the prometheus version is changed to "app.kubernetes.io/version=2.42.0" and the route is returning "404 page not found".
It was working in version "app.kubernetes.io/version=2.32.1"

**Describe the bug**

**Version**

    4.13.0-0.okd-2023-05-22-052007, UPI vsphere
**How reproducible**

Launch the prometheus route
 
**Log bundle**

**Comment by vrutkovs** [comment]
Please attach (or upload to the public file sharing service) [must-gather](https://docs.okd.io/latest/support/gathering-cluster-data.html#support_gathering_data_gathering-cluster-data) archive

**Comment by imdmahajankanika** [comment]
[must-gather.local.7153122935171610678.zip](https://github.com/okd-project/okd/files/12302828/must-gather.local.7153122935171610678.zip)

**Comment by aleskandro** [comment]
If you're referring to the Prometheus UI, it is not accessible by default anymore since a few versions. See https://github.com/okd-project/okd/discussions/1420#discussioncomment-4251967.

**Reply by imdmahajankanika** [reply]
@aleskandro Thanks! Yes, I am referring to the Prometheus UI. Do you have any idea is it related to OKD upgrade or I should raise a bug to Prometheus operator?

**Reply by aleskandro** [reply]
Hi, this is not to be considered a bug. Although uncomfortable, the UI has been deprecated in 4.10 as reported in the [4.10 ocp release notes](https://docs.openshift.com/container-platform/4.10/release_notes/ocp-4-10-release-notes.html#ocp-4-10-monitoring-prometheus):

> - OpenShift Container Platform cluster administrators can now configure query logging for Prometheus.
> - Access to the third-party Prometheus web user interface is deprecated and will be removed in a future OpenShift Container Platform release.

Also, 4.12 OCP reports the removal in [\[1\]](https://docs.openshift.com/container-platform/4.12/release_notes/ocp-4-12-release-notes.html#ocp-4-12-monitoring-prometheus-and-grafana-uis-removed) and [\[2\]](https://docs.openshift.com/container-platform/4.12/release_notes/ocp-4-12-release-notes.html#monitoring-deprecated-and-removed-features).

Depending on your use case, you might expose the thanos-querier UI via temporary solutions like `oc port-forward`.

**Reply by imdmahajankanika** [reply]
@aleskandro Thank You for the info! I tried port-forwarding but it does the redirection towards openshift login, as the oauth provider is openshift.

**Reply by aleskandro** [reply]
you should do the redirection on port 9090 (and you will not be "oauth-proxied").

**Reply by aleskandro** [reply]
this works (bypassing authentication):

```bash
oc port-forward -n openshift-monitoring deployment/thanos-querier 9090
```

Alternatively, I used this hack to access the thanos UI and get oauth working as before (users bound to the cluster-monitoring-view role will be able to access it). 

It is not thoroughly tested and shouldn't be considered secure.

After deploying the manifest, you should get a route at `https://thanos-proxy.<cluster-name>.<base-domain>`.

[expose-prometheus.yaml](https://github.com/okd-project/okd/files/12312973/expose-prometheus.yaml.txt)

**Reply by imdmahajankanika** [reply]
@aleskandro Thank You :)
